includes/board.h

```diff
@@ -13,18 +13,19 @@ class Board {
     };
 
     struct BoardCell {
-        Block& block;
+        Block* block;
         bool isOccupied;
-        BoardCell(Block& b, bool state = false) : block(b) {
-            this->isOccupied = state;
+        BoardCell() {
+            isOccupied = false;
+            block = new Block();
         }
-        void setOccupied() {
-            this->isOccupied = true;
+        BoardCell(Block& b, bool state = false) : block(&b) {
+            this->isOccupied = state;
         }
         void operator=(BoardCell& src) {
-            Block newBlock = src.block;
-            sf::Vector2f currPosition = this->block.getPos();
-            newBlock.setPos(currPosition);
+            Block* newBlock = src.block;
+            sf::Vector2f currPosition = this->block->getPos();
+            newBlock->setPos(currPosition);
```

Comment:
shouldn't this be

```
sf::Vector2f currPosition = newBlock->getPos();
this->block->setPos(currPosition);
```
or am I missing something here?

Reply:
I'm replacing the this->block with newBlock. Forgot to replace.